### PR TITLE
Add test on operate, calculate and snapshot on all components

### PR DESCRIPTION
**Pull request by koachcode**
## Unit and React Component Testing
1. Made use of React Renderer
2. Use jest testing
3. Fixed linters

### CODE REVIEW

File: src/logic/__test__/calculator-operator.test.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
+// import { render, screen } from '@testing-library/react';
+import calculator from '../calculate';
+import operate from '../operate';
```
Comment (Lameck1) [inline src/logic/__test__/calculator-operator.test.js]: ### SUGGESTION
It's a good practice to write tests separately for every single file. Please improve on this.

File: src/logic/__test__/calculator-operator.test.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+      operation: '+',
+    };
+    const buttonName = '=';
+
+    const response = calculator(obj, buttonName);
+
+    expect(response.total).toBe('7');
+    expect(response.next).toBe(null);
+  });
+});
```
Comment (Lameck1) [inline src/logic/__test__/calculator-operator.test.js]: - [ ] Well done for writing tests for the `calculate.js`. However, more tests are required. Remember we are writing unit tests and every operation in the calculate.js needs to be tested. Please include missing tests.

File: src/logic/__test__/calculator-operator.test.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  test('operation for Subtraction', () => {
+    const numOne = 6;
+    const numTwo = 3;
+    const operation = '-';
+
+    const response = operate(numOne, numTwo, operation);
+
+    expect(response).toBe('3');
+  });
+});
```
Comment (Lameck1) [inline src/logic/__test__/calculator-operator.test.js]: - [ ] Same issue as above, please include tests for other operations too.

- [ ] One last thing, you've not used React Testing testing library anywhere to test/simulate user interactions. This is part of the project requirements and as quoted below:

  > - Use React Testing Library to simulate user interaction.

  ## Suggestion
  One user interaction we can implement is clicking a combination of keys and lastly the (=) key using the `fireEvent` method from `@testing-library/react` and then asserting that the result of that calculation is present on the page using the `getByText` method.
  
  This simple [GUIDE](https://www.pluralsight.com/guides/simulate-browser-events-react-react-testing-library) from Pluralsight can really help with this.

**Review by Lameck1 — CHANGES_REQUESTED** [review]
Hi, @Ginohmk, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/logic/__test__/calculator-operator.test.js (hunk 1 above)
Comment (koachcode) [reply src/logic/__test__/calculator-operator.test.js]: Alright I will work on that, thanks

File: src/logic/__test__/calculator-operator.test.js (hunk 3 above)
Comment (koachcode) [reply src/logic/__test__/calculator-operator.test.js]: Thanks for the guide

**Review by moise10r — APPROVED** [review]
<h1>Status: Approve  :champagne:  :+1: :confetti_ball: 

### **Hi @TEAM** , 

### **Congratulations! 🎉 Project Approved ✅**

Great job implementing changes required by the previous reviewer :+1: 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 



## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers :clinking_glasses:  and Happy coding!!! :dancers:💻  

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me @moise10r when you comment so I can receive the notification.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**